### PR TITLE
Task/date helper

### PR DESCRIPTION
**Pull request by BHillenbrand722**
Added extra functionality to DateHelper.

### CODE REVIEW

**Comment by BHillenbrand722** [comment]
@ivakegg @ejrgilbert @4a4f484e I am unable to request reviews, however, I would greatly appreciate getting this PR reviewed.

File: src/main/java/datawave/util/time/DateHelper.java [hunk 1]
```diff
@@ -121,45 +155,153 @@ public static String formatToTimeExactToSeconds(long inMillis) {
...
+     *
+     * @param date
+     * @return
+     */
+    public static String formatToDetails(Date date) {
+        return DTF_Details.format(date.toInstant());
+    }
+    
+    /**
+     * Return a string representing the given date in EEE MM d HH:mm:ss zzz yyyy format in a consistent way not dependent on local settings for calendar,
```
Comment (ejrgilbert) [inline src/main/java/datawave/util/time/DateHelper.java]:
```suggestion
     * Return a string representing the given date in yyyy-MM-dd'T'HH:mm:ss.SSSZ format in a consistent way not dependent on local settings for calendar,
```

File: src/main/java/datawave/util/time/DateHelper.java [hunk 2]
```diff
@@ -340,4 +492,17 @@ public static boolean dateAtHour(Date date, int hour) {
...
+     * Custom Exception for when DateHelper cannot parse a given date within Datawave
+     */
+    public static class DatawaveDateTimeParseException extends Exception {
+        public DatawaveDateTimeParseException(String msg) {
+            super(msg);
+        }
+        
+        public DatawaveDateTimeParseException(String msg, Throwable cause) {
+            super(msg, cause);
+        }
```
Comment (ejrgilbert) [inline src/main/java/datawave/util/time/DateHelper.java]: Is there a reason for wrapping the exception in a custom Datawave exception?

File: src/main/java/datawave/util/time/DateHelper.java [hunk 3]
```diff
@@ -121,45 +155,153 @@ public static String formatToTimeExactToSeconds(long inMillis) {
...
-            return Date.from(LocalDate.parse(lenientDate, parser).atStartOfDay(parser.getZone()).toInstant());
+        try {
+            String lenientDate = convertToLenient(date, formatStr);
+            if (hasTime) {
+                return Date.from(ZonedDateTime.parse(lenientDate, parser).toInstant());
+            } else {
+                return Date.from(LocalDate.parse(lenientDate, parser).atStartOfDay(parser.getZone()).toInstant()); // Fails testFailIfMissingDigits test
+            }
+        } catch (DatawaveDateTimeParseException e) {
+            e.printStackTrace();
```
Comment (ivakegg) [inline src/main/java/datawave/util/time/DateHelper.java]: printing a stack trace to stderr is not appropriate.  Firstly we should be logging this.  Secondly, why aren't we letting the exception propogate up?

**Pull request closed without merging**